internal/core/input.rs

```diff
@@ -472,6 +473,62 @@ pub enum FocusEvent {
     WindowLostFocus,
 }
 
+/// This state is used to count the clicks in the `click_interval` from the `PLATFORM_INSTANCE`.
+#[derive(Default)]
+pub struct ClickState {
+    click_count_time_stamp: Cell<Option<crate::animations::Instant>>,
+    click_count: Cell<u8>,
+}
+
+impl ClickState {
+    /// Resets the timer and count.
+    pub fn restart(&self) {
+        self.click_count.set(1);
+        self.click_count_time_stamp.set(Some(crate::animations::Instant::now()));
+    }
+
+    /// Check if the click is repeated.
+    pub fn check_repeat(&self, mouse_event: MouseEvent) -> MouseEvent {
+        match mouse_event {
+            MouseEvent::Pressed { position, button, .. } => {
+                let instant_now = crate::animations::Instant::now();
+
+                if let Some(click_count_time_stamp) = self.click_count_time_stamp.get() {
+                    if instant_now - click_count_time_stamp
+                        < crate::platform::PLATFORM_INSTANCE
+                            .with(|p| p.get().map(|p| p.click_interval()))
+                            .unwrap_or_default()
+                        && button == PointerEventButton::Left
+                    {
+                        self.click_count.set(self.click_count.get() + 1);
+                        self.click_count_time_stamp.set(Some(instant_now));
+                    } else {
+                        self.restart();
+                    }
+                } else {
+                    self.restart();
+                }
+
```

Comment:
I still think we should set the click_count_time_stamp here and not in released

Reply:
I agree to keeping the three.